src/server/api/routers/github-router.ts

```diff
@@ -0,0 +1,82 @@
+import { z } from "zod";
+import { App } from "octokit";
+import { readFileSync } from "fs";
+
+import {
+    createTRPCRouter,
+    protectedProcedure,
+    publicProcedure,
+} from "~/server/api/trpc";
+
+// GitHub router
+export const gitHubRouter = createTRPCRouter({
+    hello: publicProcedure
+        .input(z.object({ text: z.string() }))
+        .query(({ input }) => {
+            return {
+                greeting: `Hello ${input.text}`,
+            };
+        }),
+
+    // Webhook endpoint
+    makeWebhook: protectedProcedure
+        .input(z.object({
+            repo: z.string()
+        }))
+        .mutation(async ({ ctx, input }) => {
+            const { repo } = input;
+            const owner = ctx.session?.user.name;
+
+            const privatePem = readFileSync("private-key.pem", {
+                encoding: "utf-8",
+            });
+        
+            const app = new App({
+                appId: "892635",
+                privateKey: privatePem,
+            });
+        
+            const response = await app.octokit.request(
+                `GET /repos/${owner}/${repo}/installation`,
+                {
+                    owner: owner,
+                    repo: repo,
+                    headers: {
+                        "X-GitHub-Api-Version": "2022-11-28",
+                    },
+                },
+            );
+        
+            const installationId = response.data.id;
+        
+            const octokit = await app.getInstallationOctokit(installationId);
+        
+            try {
+                const response = await octokit.request(
+                    `POST /repos/${owner}/${repo}/hooks`,
+                    {
+                        owner: owner,
+                        repo: repo,
+                        name: "web",
+                        active: true,
+                        events: ["push", "pull_request"],
+                        config: {
+                            // to be replaced with actual webhook URL
+                            url: "https://example.com/webhook",
```

Comment:
Also the root might need to be an env var and then the path might need to be a constant stored somewhere.